### PR TITLE
Update download link and include in demo readme

### PR DESCRIPTION
**Pull request by kjun9**
This adds the demo download instructions to the demo readme (both on github and readthedocs)

Follow-up to address #1202 raised in https://github.com/stellargraph/stellargraph/pull/1377#pullrequestreview-400801873

Download link per notebook still needs to be done #1390 (#1460 )

readthedocs link: https://stellargraph--1459.org.readthedocs.build/en/1459/demos/index.html#download-the-demos

### CODE REVIEW

File: docs/demos/index.txt [hunk 1]
```diff
@@ -260,6 +260,21 @@ Find a demo for an algorithm
 
 See :doc:`the root README <../README>` or each algorithm's documentation for the relevant citation(s).
 
+
+Download the demos
+------------------
+
+You can run download a local copy of the demos using the :code:`curl` command below:
```
Comment (huonw) [inline docs/demos/index.txt:267]: No change required, but for future reference, I think code/literal formatting can be more conveniently expressed with just double backticks?

```suggestion
You can run download a local copy of the demos using the ``curl`` command below:
```

(I guess it's slightly different, in that the `:code:` role ones end up with the `code` CSS class in addition to `docutils literal notranslate`. 🤷‍♂️ )

File: demos/README.md [hunk 2]
```diff
@@ -280,3 +280,16 @@ The demo notebooks can be run without any installation of Python by using Binder
 <!-- DEMO TABLE MARKER -->
 
 See [the root README](../README.md) or each algorithm's documentation for the relevant citation(s).
+
+## Download the demos
```
Comment (huonw) [inline demos/README.md:284]: FWIW, https://github.com/stellargraph/stellargraph/pull/1433 removes content like this from this README. Do you think this is still valuable to include even with a switch to RtD for most things? (#1433 is designed to allow custom content like this.)

Comment (timpitman) [reply demos/README.md:284]: I think it's still useful to know how to download all of the demos; downloading each one separately from readthedocs is less convenient in many cases.

Comment (kjun9) [reply demos/README.md:284]: Ah I see, if we're linking to RTD, I don't think "downloading demos" is a particularly important custom content to add. Should I remove it from this PR?

File: docs/demos/index.txt (hunk 1 above)
Comment (kjun9) [reply docs/demos/index.txt:267]: Ah thanks, yeah I'm still having to google every little thing I wanna do in rst

File: demos/README.md (hunk 2 above)
Comment (kjun9) [reply demos/README.md:284]: > I think it's still useful to know how to download all of the demos

It'll be included in the readthedocs `demos/index.rst` still, but the question is around whether we also want to include it in `demos/README` right?

Comment (timpitman) [reply demos/README.md]: Aaah, I see.  After looking at the output, I think just the base README is enough.

Comment (huonw) [reply demos/README.md]: > is that being fixed in a different PR @huonw?

Yup: https://github.com/stellargraph/stellargraph/pull/1448 (https://stellargraph--1448.org.readthedocs.build/en/1448/README.html#getting-started)

**Pull request merged**